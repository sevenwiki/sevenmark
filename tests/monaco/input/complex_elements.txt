{{{#style="color:red" #size="14px"
Styled element with parameters
}}}

{{{#table
[[[[Cell 1,1]] [[Cell 1,2]]]]
[[[[Cell 2,1]] [[Cell 2,2]]]]
}}}

{{{#code #lang=javascript
console.log("Code block");
function test() { return 42; }
}}}