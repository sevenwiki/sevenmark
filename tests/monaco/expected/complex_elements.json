[
  {
    "StyledElement": {
      "content": [
        {
          "Text": {
            "content": "Styled element with parameters",
            "location": {
              "end_column": 31,
              "end_line": 2,
              "start_column": 1,
              "start_line": 2
            }
          }
        },
        {
          "Text": {
            "content": "\n",
            "location": {
              "end_column": 1,
              "end_line": 3,
              "start_column": 31,
              "start_line": 2
            }
          }
        }
      ],
      "location": {
        "end_column": 4,
        "end_line": 3,
        "start_column": 1,
        "start_line": 1
      },
      "parameters": {
        "size": {
          "key": "size",
          "location": {
            "end_column": 1,
            "end_line": 2,
            "start_column": 23,
            "start_line": 1
          },
          "value": [
            {
              "Text": {
                "content": "14px",
                "location": {
                  "end_column": 34,
                  "end_line": 1,
                  "start_column": 30,
                  "start_line": 1
                }
              }
            }
          ]
        },
        "style": {
          "key": "style",
          "location": {
            "end_column": 23,
            "end_line": 1,
            "start_column": 4,
            "start_line": 1
          },
          "value": [
            {
              "Text": {
                "content": "color:red",
                "location": {
                  "end_column": 21,
                  "end_line": 1,
                  "start_column": 12,
                  "start_line": 1
                }
              }
            }
          ]
        }
      }
    }
  },
  {
    "Text": {
      "content": "\n",
      "location": {
        "end_column": 1,
        "end_line": 4,
        "start_column": 4,
        "start_line": 3
      }
    }
  },
  {
    "Text": {
      "content": "\n",
      "location": {
        "end_column": 1,
        "end_line": 5,
        "start_column": 1,
        "start_line": 4
      }
    }
  },
  {
    "TableElement": {
      "content": [
        {
          "inner_content": [
            {
              "content": [
                {
                  "Text": {
                    "content": "Cell 1",
                    "location": {
                      "end_column": 11,
                      "end_line": 6,
                      "start_column": 5,
                      "start_line": 6
                    }
                  }
                },
                {
                  "Text": {
                    "content": ",",
                    "location": {
                      "end_column": 12,
                      "end_line": 6,
                      "start_column": 11,
                      "start_line": 6
                    }
                  }
                },
                {
                  "Text": {
                    "content": "1",
                    "location": {
                      "end_column": 13,
                      "end_line": 6,
                      "start_column": 12,
                      "start_line": 6
                    }
                  }
                }
              ],
              "parameters": {}
            },
            {
              "content": [
                {
                  "Text": {
                    "content": "Cell 1",
                    "location": {
                      "end_column": 24,
                      "end_line": 6,
                      "start_column": 18,
                      "start_line": 6
                    }
                  }
                },
                {
                  "Text": {
                    "content": ",",
                    "location": {
                      "end_column": 25,
                      "end_line": 6,
                      "start_column": 24,
                      "start_line": 6
                    }
                  }
                },
                {
                  "Text": {
                    "content": "2",
                    "location": {
                      "end_column": 26,
                      "end_line": 6,
                      "start_column": 25,
                      "start_line": 6
                    }
                  }
                }
              ],
              "parameters": {}
            }
          ],
          "parameters": {}
        },
        {
          "inner_content": [
            {
              "content": [
                {
                  "Text": {
                    "content": "Cell 2",
                    "location": {
                      "end_column": 11,
                      "end_line": 7,
                      "start_column": 5,
                      "start_line": 7
                    }
                  }
                },
                {
                  "Text": {
                    "content": ",",
                    "location": {
                      "end_column": 12,
                      "end_line": 7,
                      "start_column": 11,
                      "start_line": 7
                    }
                  }
                },
                {
                  "Text": {
                    "content": "1",
                    "location": {
                      "end_column": 13,
                      "end_line": 7,
                      "start_column": 12,
                      "start_line": 7
                    }
                  }
                }
              ],
              "parameters": {}
            },
            {
              "content": [
                {
                  "Text": {
                    "content": "Cell 2",
                    "location": {
                      "end_column": 24,
                      "end_line": 7,
                      "start_column": 18,
                      "start_line": 7
                    }
                  }
                },
                {
                  "Text": {
                    "content": ",",
                    "location": {
                      "end_column": 25,
                      "end_line": 7,
                      "start_column": 24,
                      "start_line": 7
                    }
                  }
                },
                {
                  "Text": {
                    "content": "2",
                    "location": {
                      "end_column": 26,
                      "end_line": 7,
                      "start_column": 25,
                      "start_line": 7
                    }
                  }
                }
              ],
              "parameters": {}
            }
          ],
          "parameters": {}
        }
      ],
      "location": {
        "end_column": 4,
        "end_line": 8,
        "start_column": 1,
        "start_line": 5
      },
      "parameters": {}
    }
  },
  {
    "Text": {
      "content": "\n",
      "location": {
        "end_column": 1,
        "end_line": 9,
        "start_column": 4,
        "start_line": 8
      }
    }
  },
  {
    "Text": {
      "content": "\n",
      "location": {
        "end_column": 1,
        "end_line": 10,
        "start_column": 1,
        "start_line": 9
      }
    }
  },
  {
    "CodeElement": {
      "content": [
        {
          "Text": {
            "content": "=javascript\nconsole.log(\"Code block\");\nfunction test() ",
            "location": {
              "end_column": 17,
              "end_line": 12,
              "start_column": 15,
              "start_line": 10
            }
          }
        },
        {
          "Text": {
            "content": "{",
            "location": {
              "end_column": 18,
              "end_line": 12,
              "start_column": 17,
              "start_line": 12
            }
          }
        },
        {
          "Text": {
            "content": " return 42; ",
            "location": {
              "end_column": 30,
              "end_line": 12,
              "start_column": 18,
              "start_line": 12
            }
          }
        },
        {
          "Text": {
            "content": "}",
            "location": {
              "end_column": 31,
              "end_line": 12,
              "start_column": 30,
              "start_line": 12
            }
          }
        },
        {
          "Text": {
            "content": "\n",
            "location": {
              "end_column": 1,
              "end_line": 13,
              "start_column": 31,
              "start_line": 12
            }
          }
        }
      ],
      "language": [],
      "location": {
        "end_column": 4,
        "end_line": 13,
        "start_column": 1,
        "start_line": 10
      }
    }
  }
]